### PR TITLE
Angular auth services

### PR DESCRIPTION
**Pull request by adamchlebek**

### CODE REVIEW

**Review by dyl10s — COMMENTED** [review]
Just wondering why interfaces here. Don't we then have to implement them? Maybe interfaces are different in typescript and I am just not understanding.

**Review by dyl10s — COMMENTED** [review]
Also, I am thinking we should expand the scope of these service stories. I originally thought this was going to be basically making the designs functional aka all the typescript logic coding behind the designs. If that's not the case here, we will need to create additional stories for that.

**Comment by adamchlebek** [comment]
> Just wondering why interfaces here. Don't we then have to implement them? Maybe interfaces are different in typescript and I am just not understanding.

For interfaces in TypeScript, this is just used for structuring. Here is a good link that explains it:

[StackOverflow](https://stackoverflow.com/questions/51716808/when-use-a-interface-or-class-in-typescript)

**Comment by adamchlebek** [comment]
> Also, I am thinking we should expand the scope of these service stories. I originally thought this was going to be basically making the designs functional aka all the typescript logic coding behind the designs. If that's not the case here, we will need to create additional stories for that.

I can keep this PR open and wait for the front end changes. I forgot about adding all the logic behind it. For now, we can keep this open and wait for design stuff to be finished.

**Comment by dyl10s** [comment]
> > Just wondering why interfaces here. Don't we then have to implement them? Maybe interfaces are different in typescript and I am just not understanding.
> 
> For interfaces in TypeScript, this is just used for structuring. Here is a good link that explains it:
> 
> [StackOverflow](https://stackoverflow.com/questions/51716808/when-use-a-interface-or-class-in-typescript)

Interesting kinda the nature of js your basically implementing them when you pass them. Cool!

File: Frontend/src/app/auth/pages/register/register.component.ts [hunk 1]
```diff
@@ -7,9 +12,46 @@ import { Component, OnInit } from '@angular/core';
...
+      Password: registerForm.password, 
+      Name: `${registerForm.firstName.trim()} ${registerForm.lastName.trim()}` })
+    .subscribe((response: GenericResponseDTO) => {
+      if(response.success) {
+        // Handle Successful Registration
+        this.router.navigate(['/auth/login']);
+      } else {
+        this.error = response.message;
+      }
+    });
```
Comment (dyl10s) [inline Frontend/src/app/auth/pages/register/register.component.ts:62]: Need some error message here if there is a failure to connect to the API

File: Frontend/src/app/auth/pages/login/login.component.ts [hunk 2]
```diff
@@ -7,9 +11,42 @@ import { Component, OnInit } from '@angular/core';
...
+        // Handle Successful Login
+        this.JwtService.setTokens(response.data.accessToken, response.data.refreshToken);
+  
+        // TODO: Remove alert and route to dashboard
+        alert(`You have been logged in as ${this.JwtService.decode().given_name}!`);
+      } else {
+        this.loginForm.controls['password'].reset();
+        this.error = response.message;
+      }
+    });
```
Comment (dyl10s) [inline Frontend/src/app/auth/pages/login/login.component.ts:57]: Need an error here if there is a failure to connect to the api

**Review by dyl10s — CHANGES_REQUESTED** [review]
In slow network conditions it can be a bit weird if there is no loading indicator, could you make the buttons turn into spinners while the API request is made? Or if you have another solution that's fine too. Just something to let the user know the page is loading.

File: Frontend/src/app/auth/pages/register/register.component.html [hunk 3]
```diff
@@ -60,7 +60,16 @@ <h4 class="text-black">Register</h4>
...
           <div class="form-group">
-            <button nbButton fullWidth status="success" (click)="register(registerForm.value)">Register</button>
+            <button 
+              nbButton 
+              fullWidth 
+              status="success" 
+              (click)="register(registerForm.value)"
+              [nbSpinner]="isLoading"
+              [disabled]="isLoading"
+              nbSpinnerStatus="danger">
```
Comment (dyl10s) [inline Frontend/src/app/auth/pages/register/register.component.html]: I feel like green would look better here than red, what do you think? Red is a scary color.

File: Frontend/src/app/auth/pages/login/login.component.html [hunk 4]
```diff
@@ -30,7 +30,16 @@ <h4 class="text-black">Login</h4>
...
           <div class="form-group">
-            <button nbButton fullWidth status="success" (click)="login(loginForm.value)">Login</button>
+            <button 
+              nbButton 
+              fullWidth 
+              status="success" 
+              (click)="login(loginForm.value)"
+              [nbSpinner]="isLoading"
+              [disabled]="isLoading"
+              nbSpinnerStatus="danger">
```
Comment (dyl10s) [inline Frontend/src/app/auth/pages/login/login.component.html]: I feel like green would look better here than red, what do you think? Red is a scary color.

**Review by dyl10s — APPROVED** [review]

**Pull request merged**